### PR TITLE
Features/8393

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
## Status
**Ready**

## Fixes or Implements [AB#8393](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8393)
- [x] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
Adds the resend phone verification and verification input to profile page.

## Testing
- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

### UI Changes
YES

### Browsers Tested
- [x] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Tests to be updated once backend is completed.

### CODE REVIEW

File: Apps/WebClient/src/Server/Controllers/UserProfileController.cs [hunk 1]
```diff
@@ -303,6 +331,64 @@ public async Task<IActionResult> GetUserEmailInvite(string hdid)
             }
         }
 
+        /// <summary>
```
Comment (sslaws) [inline Apps/WebClient/src/Server/Controllers/UserProfileController.cs:334]: I don't understand the purpose of this method.  The actual validation code for SMS should  never be returned to the client and the method above does validation.  Can you elaborate?

Comment (tiagograf-AOT) [reply Apps/WebClient/src/Server/Controllers/UserProfileController.cs:334]: its only used to check if the phone was validated or not, this is the part that I mention that could use some refactoring (together with email validation) the actual code/digit is not sent to the front end.

File: Apps/Common/src/Services/EmailQueueService.cs [hunk 2]
```diff
@@ -192,6 +192,7 @@ private Email ParseTemplate(EmailTemplate emailTemplate, Dictionary<string, stri
             }
 
             Email email = new Email();
+            email.Id = Guid.NewGuid();
```
Comment (sslaws) [inline Apps/Common/src/Services/EmailQueueService.cs:195]: For a new email, the database will generate the guid.

File: Apps/Database/src/Models/MessagingVerification.cs [hunk 3]
```diff
@@ -47,7 +47,7 @@ public class MessagingVerification : AuditableEntity
         /// Gets or sets the associated email that was sent for this verification.
         /// Required if the VerificationType = MessagingVerificationType.Email.
         /// </summary>
-        public Guid EmailId { get; set; }
+        public Guid? EmailId { get; set; }
```
Comment (sslaws) [inline Apps/Database/src/Models/MessagingVerification.cs:50]: This seems to have changed the column type... Do we want that?

Comment (tiagograf-AOT) [reply Apps/Database/src/Models/MessagingVerification.cs:50]: Yes, emailId is nullable now.

File: Apps/Common/src/Services/EmailQueueService.cs (hunk 2 above)
Comment (tiagograf-AOT) [reply Apps/Common/src/Services/EmailQueueService.cs:195]: this was a problem when saving the the message verification, it threw an error saying the emailId cannot be null for Email verification type.

File: Apps/Common/test/unit/Services/Email_Test.cs [hunk 4]
```diff
@@ -56,6 +56,7 @@ public void ShouldProcessTemplate()
                 new Mock<IMessagingVerificationDelegate>().Object,
                 new Mock<IWebHostEnvironment>().Object);
             Email actual = emailService.ProcessTemplate(emailTo, template, d);
+            expected.Id = actual.Id;
```
Comment (FuriousLlama) [inline Apps/Common/test/unit/Services/Email_Test.cs:59]: is this done because we dont know a-priori the id?

Comment (tiagograf-AOT) [reply Apps/Common/test/unit/Services/Email_Test.cs:59]: yep!

**Pull request merged**